cmd/frontend/auth/user.go

```diff
@@ -61,58 +61,64 @@ type GetAndSaveUserOp struct {
 // 🚨 SECURITY: The safeErrMsg is an error message that can be shown to unauthenticated users to
 // describe the problem. The err may contain sensitive information and should only be written to the
 // server error logs, not to the HTTP response to shown to unauthenticated users.
-func GetAndSaveUser(ctx context.Context, db database.DB, op GetAndSaveUserOp) (userID int32, safeErrMsg string, err error) {
+func GetAndSaveUser(ctx context.Context, db database.DB, op GetAndSaveUserOp) (newUserCreated bool, userID int32, safeErrMsg string, err error) {
 	if MockGetAndSaveUser != nil {
 		return MockGetAndSaveUser(ctx, op)
 	}
 
 	externalAccountsStore := db.UserExternalAccounts()
 	logger := sglog.Scoped("authGetAndSaveUser", "get and save user authenticated by external providers")
 
-	userID, userSaved, extAcctSaved, safeErrMsg, err := func() (int32, bool, bool, string, error) {
+	newUserCreated, userID, userSaved, extAcctSaved, safeErrMsg, err := func() (bool, int32, bool, bool, string, error) {
```

Comment:
```suggestion
	userID, userSaved, extAcctSaved, safeErrMsg, err := func() (bool, int32, bool, bool, string, error) {
```

Looking at this again, `newUserCreated` is always equal to `userSaved`. So `userSaved` tells you if a new user was created or not. No need for another return value.